File: sales-copilot-preview/src/types/followup.ts
import type { CustomerIntent, CustomerState, ObjectionType, RiskLevel, SalesStage } from './salesContext'

export type FollowupIndustry = 'windows' | 'renovation' | 'custom_furniture' | 'building_materials'

export type FollowupCustomerStatus =
  | 'new_inquiry'
  | 'comparing'
  | 'hesitating'
  | 'ready_to_close'
  | 'silent'

export type FollowupIntentLevel = '高意向' | '中意向' | '低意向' | '待判断'

export interface FollowupInput {
  industry: FollowupIndustry
  customerStatus: FollowupCustomerStatus
  transcript: string
}

export interface FollowupSignal {
  label: string
  evidence: string
}

export interface FollowupPlanOption {
  title: string
  description: string
}

export type DialogueSpeaker = 'sales' | 'customer' | 'unknown' | 'noise'
export type DialogueMessageSource = 'manual' | 'realtime_asr' | 'sales_recap' | 'wechat' | 'attachment' | 'field'

export interface DialogueMessage {
  id: string
  speaker: DialogueSpeaker
  text: string
  confidence?: number
  timestamp?: string
  source?: DialogueMessageSource
}

export interface FollowupResult {
  intentLevel: FollowupIntentLevel
  primaryConcern: string
  currentStage: string
  missedPoint: string
  nextAction: string
  wechatScript: string
  planTitle: string
  planOptions: FollowupPlanOption[]
  signals: FollowupSignal[]
  context: {
    stage: SalesStage
    customerState: CustomerState
    customerIntent: CustomerIntent
    objectionType: ObjectionType
    riskLevel: RiskLevel
    confidence: number
  }
}

File: backend/app/schemas/followup.py
from typing import Literal

from pydantic import BaseModel, Field


FollowupIndustry = Literal["windows", "renovation", "custom_furniture", "building_materials"]
FollowupCustomerStatus = Literal["new_inquiry", "comparing", "hesitating", "ready_to_close", "silent"]
FollowupQuality = Literal["standard", "premium"]
AttachmentKind = Literal["photo", "floorplan", "quote"]


class FollowupGenerateRequest(BaseModel):
    industry: FollowupIndustry
    customerStatus: FollowupCustomerStatus
    transcript: str = Field(min_length=1)
    quality: FollowupQuality = "standard"


class FollowupAttachmentAnalyzeRequest(BaseModel):
    kind: AttachmentKind
    name: str
    mimeType: str
    dataUrl: str = Field(min_length=1)
    note: str | None = None


class FollowupAttachmentAnalyzeResponse(BaseModel):
    summary: str
    model: str


class FollowupSignal(BaseModel):
    label: str
    evidence: str


class FollowupPlanOption(BaseModel):
    title: str
    description: str


class FollowupContext(BaseModel):
    stage: str
    customerState: str
    customerIntent: str
    objectionType: str
    riskLevel: str
    confidence: float = Field(ge=0, le=1)


class FollowupGenerateResponse(BaseModel):
    intentLevel: str
    primaryConcern: str
    currentStage: str
    missedPoint: str
    nextAction: str
    wechatScript: str
    planTitle: str
    planOptions: list[FollowupPlanOption]
    signals: list[FollowupSignal]
    context: FollowupContext
    model: str
    quality: FollowupQuality
